test/e2e/framework/network/utils.go

```diff
@@ -602,6 +610,35 @@ func (config *NetworkingTestConfig) createNetShellPodSpec(podName, hostname stri
 			Protocol:      v1.ProtocolSCTP,
 		})
 	}
+
+	if config.EndpointsHostNetwork {
+		pod.Spec.Containers[0].Env = []v1.EnvVar{
+			{
+				Name: "HOST_IP",
+				ValueFrom: &v1.EnvVarSource{
+					FieldRef: &v1.ObjectFieldSelector{
+						FieldPath: "status.hostIP",
+					},
+				},
+			},
+			{
+				Name: "POD_IP0",
+				ValueFrom: &v1.EnvVarSource{
+					FieldRef: &v1.ObjectFieldSelector{
+						FieldPath: "status.podIPs[0].ip",
+					},
+				},
+			},
+			{
+				Name: "POD_IP1",
+				ValueFrom: &v1.EnvVarSource{
+					FieldRef: &v1.ObjectFieldSelector{
+						FieldPath: "status.podIPs[1].ip",
```

Comment:
just curious, what happens if podIPs only has 1 element? is POD_IP1 empty?  can the downward api deal with it?

Reply:
I ingenuosly thought the env variable would be set as empty, but I will try setting it to 2 to see what happens

Reply:
Actually, not sure what I tested yesterday, but choosing an item of the array won't work.
The pod is failing with `Invalid value: "status.podIPs[0]": error converting fieldPath: field label not supported: status.podIPs[0]`

What works, is using `status.podIPs` and fills the variable with a comma separated list of addresses.